-- Found ament_cmake: 0.9.8 (/home/parham/ros2_foxy/ros2-linux/share/ament_cmake/cmake)
-- Found PythonInterp: /usr/bin/python3 (found suitable version "3.8.10", minimum required is "3") 
-- Using PYTHON_EXECUTABLE: /usr/bin/python3
-- Found ament_lint_auto: 0.9.6 (/home/parham/ros2_foxy/ros2-linux/share/ament_lint_auto/cmake)
-- Found rosidl_default_generators: 1.0.0 (/home/parham/ros2_foxy/ros2-linux/share/rosidl_default_generators/cmake)
-- Using all available rosidl_typesupport_c: rosidl_typesupport_fastrtps_c;rosidl_typesupport_introspection_c;rosidl_typesupport_connext_c
-- Found rosidl_adapter: 1.2.0 (/home/parham/ros2_foxy/ros2-linux/share/rosidl_adapter/cmake)
-- Found RTI Connext: /opt/rti.com/rti_connext_dds-5.3.1
-- Using all available rosidl_typesupport_cpp: rosidl_typesupport_fastrtps_cpp;rosidl_typesupport_introspection_cpp;rosidl_typesupport_connext_cpp
-- Found PythonInterp: /usr/bin/python3 (found version "3.8.10") 
CMake Error at /home/parham/ros2_foxy/ros2-linux/share/rosidl_adapter/cmake/rosidl_adapt_interfaces.cmake:60 (message):
  execute_process(/usr/bin/python3 -m rosidl_adapter --package-name
  push_button_interface --arguments-file
  /home/parham/Dev/ros2_foxy/science/science-gui/science_ws/build/push_button_interface/rosidl_adapter__arguments__push_button_interface.json
  --output-dir
  /home/parham/Dev/ros2_foxy/science/science-gui/science_ws/build/push_button_interface/rosidl_adapter/push_button_interface
  --output-file
  /home/parham/Dev/ros2_foxy/science/science-gui/science_ws/build/push_button_interface/rosidl_adapter/push_button_interface.idls)
  returned error code 1:

  Error processing 'size_t platform_height' of
  'push_button_interface/PushButton': ''size_t' is an invalid message name.
  It should have the pattern '^[A-Z][A-Za-z0-9]*$''

  /usr/lib/python3/dist-packages/requests/__init__.py:89:
  RequestsDependencyWarning: urllib3 (1.26.7) or chardet (3.0.4) doesn't
  match a supported version!

    warnings.warn("urllib3 ({}) or chardet ({}) doesn't match a supported "

  Traceback (most recent call last):

    File "/usr/lib/python3.8/runpy.py", line 194, in _run_module_as_main
      return _run_code(code, main_globals, None,
    File "/usr/lib/python3.8/runpy.py", line 87, in _run_code
      exec(code, run_globals)
    File "/opt/ros/foxy/lib/python3.8/site-packages/rosidl_adapter/__main__.py", line 19, in <module>
      sys.exit(main())
    File "/opt/ros/foxy/lib/python3.8/site-packages/rosidl_adapter/main.py", line 53, in main
      abs_idl_file = convert_to_idl(
    File "/opt/ros/foxy/lib/python3.8/site-packages/rosidl_adapter/__init__.py", line 19, in convert_to_idl
      return convert_msg_to_idl(
    File "/opt/ros/foxy/lib/python3.8/site-packages/rosidl_adapter/msg/__init__.py", line 28, in convert_msg_to_idl
      msg = parse_message_string(package_name, input_file.stem, content)
    File "/opt/ros/foxy/lib/python3.8/site-packages/rosidl_adapter/parser.py", line 516, in parse_message_string
      Type(type_string, context_package_name=pkg_name),
    File "/opt/ros/foxy/lib/python3.8/site-packages/rosidl_adapter/parser.py", line 276, in __init__
      super(Type, self).__init__(
    File "/opt/ros/foxy/lib/python3.8/site-packages/rosidl_adapter/parser.py", line 204, in __init__
      raise InvalidResourceName(

  rosidl_adapter.parser.InvalidResourceName: 'size_t' is an invalid message
  name.  It should have the pattern '^[A-Z][A-Za-z0-9]*$'

Call Stack (most recent call first):
  /home/parham/ros2_foxy/ros2-linux/share/rosidl_cmake/cmake/rosidl_generate_interfaces.cmake:130 (rosidl_adapt_interfaces)
  CMakeLists.txt:37 (rosidl_generate_interfaces)


-- Configuring incomplete, errors occurred!
See also "/home/parham/Dev/ros2_foxy/science/science-gui/science_ws/build/push_button_interface/CMakeFiles/CMakeOutput.log".
